client/settings/payment-methods-selector/index.js

```diff
@@ -0,0 +1,114 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Button, CheckboxControl, Icon, Modal } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { HorizontalRule } from '@wordpress/primitives';
+
+/**
+ * Internal dependencies
+ */
+import { addSelectedPaymentMethods } from 'data';
+import PaymentMethodIcon from 'settings/payment-method-icon';
+import './style.scss';
+
+const PaymentMethodsSelector = ( props ) => {
+	const { onClose, enabledPaymentMethods = [] } = props;
+	const [ paymentMethods, setPaymentMethods ] = useState(
+		enabledPaymentMethods.reduce(
+			( acc, value ) => {
+				acc[ value ] = true;
+				return acc;
+			},
+			{ giropay: false, sofort: false, sepa: false }
+		)
+	);
+
+	const onChangePaymentMethod = ( paymentMethod ) => {
+		return ( enabled ) => {
+			setPaymentMethods( {
+				...paymentMethods,
+				[ paymentMethod ]: enabled,
+			} );
+		};
+	};
+
+	const onAddSelected = () => {
+		const selectedPaymentMethods = Object.entries( paymentMethods )
+			.filter( ( [ , enabled ] ) => enabled )
+			.map( ( [ method ] ) => method );
+		addSelectedPaymentMethods( selectedPaymentMethods );
+		onClose();
```

Comment:
Not sure if this will work until we implement `addSelectedPaymentMethods`. It may be needed to trigger this through the Redux pipeline.